### ISSUE TITLE
the input_dim of embedding layer when set mask_zero = true

### ISSUE THREAD

**Issue by tttwwy** [fchollet/keras#744]
For example, my dict size is 100, when I set mask_zero = False, the input_dim of embedding layer should be 100 obviously, but when I set mask_zero = True, my input id start from 1 until the end, the input_dim should be 100 or 101?

**Comment by fchollet** [comment]
If you have 100 features and you use 0 as a masking value, it means that your feature values will go from 1 to 101. So you should use an `input_dim` of 101.

**Comment by tttwwy** [comment]
If i want to use word2vec to init the weigths matrix of embedding layer, then the word with ID = 1 should corresponding to weights[1],not weights[0], am I right?

**Comment by MaratZakirov** [comment]
mask_zero: Whether or not the input value 0 is a special "padding" value that should be masked out. This is useful for recurrent layers which may take variable length input. If this is True then all subsequent layers in the model need to support masking or an exception will be raised. If mask_zero is set to True, as a consequence, index 0 cannot be used in the vocabulary (**input_dim should equal |vocabulary| + 2**).

taken from https://keras.io/layers/embeddings/
